frontend/viewer/src/main.ts

```diff
@@ -4,9 +4,13 @@ import './lib/append-head-hack';
 import './app.postcss';
 
 import App from './App.svelte';
+import {setupServiceProvider} from '$lib/services/service-provider';
 import {setupDotnetServiceProvider} from './lib/services/service-provider-dotnet';
+import {useEventBus} from '$lib/services/event-bus';
 
+setupServiceProvider();
```

Comment:
This is the same thing, right?
```suggestion
import '$lib/services/service-provider';
import {setupDotnetServiceProvider} from './lib/services/service-provider-dotnet';
import {useEventBus} from '$lib/services/event-bus';

```

Reply:
I just change service provider to be side effect free